clipmenu

```diff
@@ -45,6 +45,9 @@ if [[ "$CM_LAUNCHER" == rofi-script ]]; then
         # shellcheck disable=SC2124
         chosen_line="${@: -1}"
     fi
+elif [[ ! -t 1 ]]; then
```

Comment:
Hmm, I think this seems good, but it should likely be implemented by a flag rather than by checking if stdout is connected to a terminal. Otherwise I worry this feature is too hard to turn off.

Reply:
You mean, like `clipmenu --print`?

Currently "all arguments are passed through to dmenu itself" (`-h`/`--help` being the exception), so it would have to be changed and maybe could get a little confusing.

Reply:
I think with the comment below, this can be done reasonably as it is in mpdmenu :-)

Reply:
So, are you suggesting we change the way arguments are handled to make it `mpdmenu`-like?

```
Pass clipmenu arguments first, followed by any dmenu arguments. They are separated by ::. For example:

clipmenu -p :: -sb '#000000'
```

Wouldn't it break current workflows?

Reply:
> it should likely be implemented by a flag rather than by checking if stdout is connected to a terminal. Otherwise I worry this feature is too hard to turn off.

I'd argue that having selection passed through a pipe or redirected to a file would be the expected behavior when using *nix tools. Why would someone run something like `clipmenu > test` and expect it to print nothing to `test` (current behavior)?

The only flaw I see in this implementation is that what is printed is the content in `line_cache`, instead of the selection itself, so manipulating the output would need some extra work. Maybe this should be changed, but it'd require some thought on how to handle multiple selections.